src/include/migraphx/op/multinomial.hpp

```diff
@@ -21,11 +21,42 @@
  * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
  * THE SOFTWARE.
  */
+
+/**
+ * Multinomial or categorical distribution.  Performs a sampling and returns a count of
+ *         each category, or bucket.  This does not require the standard multinomial
+ *         distribution but instead takes a probability distribution as an input.
+ *
+ *          In the large number limit, the fractional counts approach the multinomial distribution.
+ *
+ *      Inputs:   args[0] - a vector of probabilities for each category.  Values are running
+ *                          totals as provided by op prefix_scan_sum.
+ *                          Values are log normalized (i.e. start with any set of numbers > 0, then
+ *                          val[i] = log(val[i]) / sum (log(val[0]) + log(val[1])+ ...) )
```

Comment:
Didn't the `parse_multinomial` take the element-wise exponent of the probabilities? The probabilities would be then no longer be `log(x)`.

Reply:
This comment has been corrected.  Inputs to parse_multinomial are logs of probabilities and it adds an exp instruction to convert to non-logs.  The MigraphX Multinomial op. doesn't use the same inputs.  It takes (a cumulative sum of) straight values as inputs, but they do not have to be normalized.  Multiplying or dividing them by any constant leaves the relative probabilities unchanged.  The normalization factor is the last value of the cumulative sum and is applied at runtime.